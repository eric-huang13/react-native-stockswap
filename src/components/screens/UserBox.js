import React, {Component} from 'react';
import {Text, View, StyleSheet, Image, SafeAreaView} from 'react-native';

export default class UserBox extends Component {
  render() {
    const {item} = this.props;

    return (
      <SafeAreaView
        style={
          item.id % 2 === 0
            ? {...style.container, backgroundColor: '#2a334a'}
            : {...style.container}
        }>
        <Image style={style.image} source={{uri: item.img}} />

        <View style={style.detailsContainer}>
          <Text style={style.name}>{item.name}</Text>

          <Text style={style.posts}>
            Posts: <Text style={style.detailsBold}>{item.posts}</Text>
          </Text>
          <Text style={style.followers}>
            Followers: <Text style={style.detailsBold}>{item.followers}</Text>
          </Text>
          <Text style={style.followers}>
            Trading since: <Text style={style.detailsBold}>{item.followers}</Text>
          </Text>
        </View>
        <View style={style.percentageView}>
          <Text style={style.percentage}>{item.percentage}%</Text>
        </View>
      </SafeAreaView>
    );
  }
}

const style = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingVertical: 14,
    paddingHorizontal: 10,
    backgroundColor: '#324165',
  },
  image: {
    height: 63,
    width: 63,
    borderRadius: 50,
  },
  imgDetailContainer: {
    flexDirection: 'row',
    marginLeft: -8,
  },
  detailsContainer: {
    flexDirection: 'column',
    marginLeft: -150,
    justifyContent: 'space-between',
  },
  name: {
    fontSize: 16,
    color: '#FFFFFF',
    fontFamily: 'Montserrat-Bold',
  },
  posts: {
    color: 'lightgrey',
    fontFamily: 'Montserrat-Regular',
    fontSize: 12,
  },
  followers: {
    color: 'lightgrey',
    fontFamily: 'Montserrat-Regular',
    fontSize: 12,
  },
  percentageView: {
    flexDirection: 'row',
    alignContent: 'center',
  },
  percentage: {
    alignSelf: 'center',
    color: '#1AB968',
    fontSize: 16,
    fontFamily: 'Montserrat-Bold',
  },
  detailsBold: {
    fontFamily: 'Montserrat-Medium',
    color: '#FFFFFF',
    fontSize: 12,
  },
});
