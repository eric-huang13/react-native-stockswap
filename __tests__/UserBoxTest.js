import React from 'react';
import UserBox from '../src/components/SearchTabComponents/UserBox';
import ShallowRenderer from 'react-test-renderer/shallow'; 


jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock"),
);
jest.mock('react-redux', () => ({
    connect: () => {
      return (component) => {
        return component
      };
    }
  }));

test('renders correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<UserBox item={{img:'test' }}/>);
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});