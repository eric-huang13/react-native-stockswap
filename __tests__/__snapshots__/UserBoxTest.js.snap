// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "backgroundColor": "#2a334a",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 14.124087591240876,
      "paddingVertical": 19.77372262773723,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "test",
      }
    }
    style={
      Object {
        "borderRadius": 70.62043795620437,
        "height": 88.98175182481751,
        "width": 88.98175182481751,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "column",
        "justifyContent": "space-between",
        "marginLeft": -211.86131386861314,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Montserrat-Bold",
          "fontSize": 22.598540145985403,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "lightgrey",
          "fontFamily": "Montserrat-Regular",
          "fontSize": 16.94890510948905,
        }
      }
    >
      Posts: 
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Montserrat-Medium",
            "fontSize": 16.94890510948905,
          }
        }
      />
    </Text>
    <Text
      style={
        Object {
          "color": "lightgrey",
          "fontFamily": "Montserrat-Regular",
          "fontSize": 16.94890510948905,
        }
      }
    >
      Followers: 
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Montserrat-Medium",
            "fontSize": 16.94890510948905,
          }
        }
      />
    </Text>
    <Text
      style={
        Object {
          "color": "lightgrey",
          "fontFamily": "Montserrat-Regular",
          "fontSize": 16.94890510948905,
        }
      }
    >
      Trading since:
       
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Montserrat-Medium",
            "fontSize": 16.94890510948905,
          }
        }
      />
    </Text>
  </View>
  <View
    style={
      Object {
        "alignContent": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#1AB968",
          "fontFamily": "Montserrat-Bold",
          "fontSize": 22.598540145985403,
        }
      }
    >
      %
    </Text>
  </View>
</ForwardRef(SafeAreaView)>
`;
